core/embed/trezorhal/image.c

```diff
@@ -50,35 +50,71 @@ static secbool compute_pubkey(uint8_t sig_m, uint8_t sig_n,
   return sectrue * (0 == ed25519_cosi_combine_publickeys(res, keys, sig_m));
 }
 
-secbool load_image_header(const uint8_t *const data, const uint32_t magic,
-                          const uint32_t maxsize, uint8_t key_m, uint8_t key_n,
-                          const uint8_t *const *keys, image_header *const hdr) {
+secbool parse_image_header(const uint8_t *const data, const uint32_t magic,
```

Comment:
yeah i didn't want to introduce the key check in firmware - it might, however, prevent a disaster if we in some weird state of mind release signed production firmware with unsigned bootloader and no other checks nor QA will catch that 

will try to rename things to make more sense.

Reply:
renamed and reshuffled:
https://github.com/trezor/trezor-firmware/commit/df3eccea7ad47dae23ec6f4c1d67990f19a3e39b

some basic checks were left in the `read_image_header` functions, since additional separation bloats the code elsewhere.

for vendor header the changes are smaller, left basically `read_vendor_header` with checks and `check_vendor_header_sig` for checking signature